examples/src/big_compute_buffers/mod.rs

```diff
@@ -0,0 +1,288 @@
+
+    let bind_group_layout = device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+        label: Some("Custom Storage Bind Group Layout"),
+        entries: &bind_group_layout_entries,
+    });
+
+    let bind_group = device.create_bind_group(&wgpu::BindGroupDescriptor {
+        label: Some("Combined Storage Bind Group"),
+        layout: &bind_group_layout,
+        entries: &bind_group_entries,
+    });
+    (bind_group_layout, bind_group)
+}
+
+fn calculate_chunks(numbers: &[f32], max_buffer_size: u64) -> Vec<&[f32]> {
+    let max_elements_per_chunk = max_buffer_size as usize / std::mem::size_of::<f32>(); // Calculate max f32 elements per buffer
+    numbers.chunks(max_elements_per_chunk).collect()
+}
+
+fn create_storage_buffers(
+    device: &wgpu::Device,
+    numbers: &[f32],
+    input_size: u64, // bytes..
+) -> Vec<wgpu::Buffer> {
+    if input_size > MAX_BUFFER_SIZE {
+        let chunks = calculate_chunks(numbers, MAX_BUFFER_SIZE);
+
+        chunks
+            .iter()
+            .enumerate()
+            .map(|(e, seg)| {
+                device.create_buffer_init(&wgpu::util::BufferInitDescriptor {
+                    label: Some(&format!("Storage Buffer-{}", e)),
+                    contents: bytemuck::cast_slice(seg),
+                    usage: wgpu::BufferUsages::STORAGE
+                        | wgpu::BufferUsages::COPY_DST
+                        | wgpu::BufferUsages::COPY_SRC,
+                })
+            })
+            .collect()
+    } else {
+        vec![
+            device.create_buffer_init(&wgpu::util::BufferInitDescriptor {
+                label: Some("Storage Buffer-0"),
+                contents: bytemuck::cast_slice(numbers),
+                usage: wgpu::BufferUsages::STORAGE
+                    | wgpu::BufferUsages::COPY_DST
+                    | wgpu::BufferUsages::COPY_SRC,
+            }),
+        ]
+    }
+}
+
+fn create_staging_buffers(device: &wgpu::Device, numbers: &[f32]) -> Vec<wgpu::Buffer> {
+    let chunks = calculate_chunks(numbers, MAX_BUFFER_SIZE);
+
+    (0..chunks.len())
+        .map(|e| {
+            let size = std::mem::size_of_val(chunks[e]) as u64;
+
+            device.create_buffer(&wgpu::BufferDescriptor {
+                label: Some(&format!("staging buffer-{}", e)),
+                size,
+                usage: wgpu::BufferUsages::MAP_READ | wgpu::BufferUsages::COPY_DST,
+                mapped_at_creation: false,
+            })
+        })
+        .collect()
+}
+
+async fn run() {
+    let numbers = {
+        let bytes_per_gb = 1024 * 1024 * 1024;
+        let bytes_per_f32 = std::mem::size_of::<f32>();
+        let total_bytes = (0.25 * bytes_per_gb as f32) as usize;
+        let elements = total_bytes / bytes_per_f32;
+
+        vec![0.0; elements]
+    };
+    assert!(numbers.iter().all(|n| *n == 0.0));
+    log::info!("All 0.0s");
+
+    let t1 = std::time::Instant::now();
+    let results = execute_gpu(&numbers).await.unwrap();
```

Comment:
Rather than having everything return an `Option` that gets unwrapped here, I think you get better stack traces if you just use `panic!` or `unreachable!`. (Clippy says you shouldn't use `panic!` in production code, but 1) this isn't production code and 2) I'm not sure I agree with Clippy in principle.)